tests/cpp_api/audio_tests/audio_tests.cpp

```diff
@@ -0,0 +1,208 @@
+            } else {
+                std::cout << "\nUnable to read all data from golden outputs\n";
+                return pass_status;
+            }
+        }
+    } else {
+        std::cout << "\nCould not open the reference output. Please check the path specified\n";
+        return pass_status;
+    }
+
+    int matched_indices = 0;
+    for (int j = 0; j < frames; j++) {
+        float ref_val, out_val;
+        ref_val = ref_output[j];
+        out_val = dst_ptr[j];
+        bool invalid_comparison = ((out_val == 0.0f) && (ref_val != 0.0f));
+        if (!invalid_comparison && abs(out_val - ref_val) < 1e-20)
+            matched_indices += 1;
+    }
+
+    std::cout << std::endl << "Results for Test case: " << std::endl;
+    if ((matched_indices == frames) && matched_indices != 0) {
+        pass_status = true;
+    }
+
+    return pass_status;
+}
+
+int test(int test_case, const char *path, int qa_mode, int downmix, int gpu);
+int main(int argc, const char **argv) {
+    // check command-line usage
+    const int MIN_ARG_COUNT = 2;
+    if (argc < MIN_ARG_COUNT) {
+        printf("Usage: ./audio_tests <audio-dataset-folder - required> <test_case> <downmix=0/1> <device-gpu=1/cpu=0> <qa_mode=0/1>\n");
+        return -1;
+    }
+
+    int argIdx = 0;
+    const char *path = argv[++argIdx];
+    int qa_mode = 0;
+    unsigned test_case = 0;
+    bool downmix = false;
+    bool gpu = 0;
+
+    if (argc >= argIdx + MIN_ARG_COUNT)
+        test_case = atoi(argv[++argIdx]);
+
+    if (argc >= argIdx + MIN_ARG_COUNT)
+        downmix = atoi(argv[++argIdx]);
+
+    if (argc >= argIdx + MIN_ARG_COUNT)
+        gpu = atoi(argv[++argIdx]);
+
+    if (argc >= argIdx + MIN_ARG_COUNT)
+        qa_mode = atoi(argv[++argIdx]);
+
+    if (gpu) {  // TODO - Will be removed when GPU support is added for Audio pipeline
+        std::cout << "WRN : Currently Audio unit test supports only HOST backend\n";
+        gpu = false;
+    }
+
+    int return_val = test(test_case, path, qa_mode, downmix, gpu);
+    return return_val;
+}
+
+int test(int test_case, const char *path, int qa_mode, int downmix, int gpu) {
+    int input_batch_size = 1;
+    bool is_output_audio_decoder = false;
+    std::cout << "test case " << test_case << std::endl;
+    std::cout << "Running on " << (gpu ? "GPU" : "CPU") << std::endl;
+
+    auto handle = rocalCreate(input_batch_size,
+                              gpu ? RocalProcessMode::ROCAL_PROCESS_GPU : RocalProcessMode::ROCAL_PROCESS_CPU, 0,
+                              1);
+
+    if (rocalGetStatus(handle) != ROCAL_OK) {
+        std::cout << "Could not create the Rocal contex\n";
+        return -1;
+    }
+
+    std::cout << "Running LABEL READER" << std::endl;
+    rocalCreateLabelReader(handle, path);
+
+    if (test_case == 0)
+        is_output_audio_decoder = true;
+    RocalTensor decoded_output = rocalAudioFileSourceSingleShard(handle, path, 0, 1, is_output_audio_decoder, false, false, downmix);
```

Comment:
decoded_output is not user and gives a warning in compilation. Please remove it or check the return value